lib/puppet/type/tcp_conn_validator.rb

```diff
@@ -0,0 +1,42 @@
+Puppet::Type.newtype(:tcp_conn_validator) do
+
+  @doc = "Verify that a TCP connection can be successfully established between a node
+          and the expected  server.  Its primary use is as a precondition to
+          prevent configuration changes from being applied if the server cannot be
+          reached, but it could potentially be used for other purposes such as monitoring."
+
+  ensurable do
```

Comment:
I agree with @adrienthebo that ensurable for this resource doesn't make a ton of sense. I'd advocate for removing it or setting up something like the 'reachable' parameter finch reccomended on the initial PR to stdlib.